users/ericgebhart/process_records.c

```diff
@@ -0,0 +1,255 @@
+/*
+  Copyright 2018 Eric Gebhart <e.a.gebhart@gmail.com>
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "ericgebhart.h"
+#include "caps_word.h"
+#include "g/keymap_combo.h"
+
+__attribute__((weak)) bool process_record_keymap(uint16_t keycode, keyrecord_t *record) { return true; }
+
+__attribute__((weak)) bool process_record_secrets(uint16_t keycode, keyrecord_t *record) { return true; }
+
+
+uint16_t tap_taplong_timer;
+
+inline void tap_taplong(uint16_t kc1, uint16_t kc2, keyrecord_t *record) {
+  if (record->event.pressed) {
+    tap_taplong_timer = timer_read();
+  } else {
+    if (timer_elapsed(tap_taplong_timer) > TAPPING_TERM) {
+      tap_code16(kc2);
+    } else {
+      tap_code16(kc1);
+    }
+  }
+}
+
+/* for (){}[]""''<>``. tap for open. Hold for open and close, ending inbetween. */
+/* Assumes a one character length.                                              */
+inline void open_openclose(uint16_t kc1, uint16_t kc2, keyrecord_t *record) {
+  if (record->event.pressed) {
+    tap_taplong_timer = timer_read();
+  }else{
+    if (timer_elapsed(tap_taplong_timer) > TAPPING_TERM) {
+      tap_code16(kc1);
+      tap_code16(kc2);
+      tap_code16(KC_LEFT);
+    } else {
+      tap_code16(kc1);
+    }
+  }
+}
+
+// Defines actions for my global custom keycodes. Defined in ericgebhart.h file
+// Then runs the _keymap's record handier if not processed here
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  // If console is enabled, it will print the matrix position and status of each key pressed
+
+#ifdef OLED_DRIVER_ENABLE
```

Comment:
Not sure if you want this as `CUSTOM_OLED_DRIVER_CODE` as per the define in your rules or the default `OLED_ENABLE`